command/runner.go

```diff
@@ -203,3 +223,67 @@ func (r *cmdWaitReadCloser) Close() error {
 	r.closed = true
 	return nil
 }
+
+// Close the pipe and wait until the consumer program exits
+func (p pagerPipe) Close() error {
+	err := p.writer.Close()
+	<-*p.cc
+	return err
+}
+
+func (p pagerPipe) Write(b []byte) (int, error) {
+	return p.writer.Write(b)
+}
+
+func runPager() (*pagerPipe, error) {
+	pager := ""
+	if p := os.Getenv("KUBECOLOR_PAGER"); p != "" {
+		pager = p
+	} else if p := os.Getenv("PAGER"); p != "" {
+		pager = p
+	} else {
+		for _, p := range DefaultPagers {
+			if _, err := exec.LookPath(p); err == nil {
+				pager = p
+				if pager == "less" {
+					pager = "less -R"
+				}
+				break
+			}
+		}
+	}
```

Comment:
Suggest extracting this "obtain pager command" into a separate function, and return on first hit.

That way you don't need all these `else` statements.

You could also make the function return `[]string{}` immediately, as extracting the fields is only needed on the user input, and not the defaults.

Also, maybe then you could write some simple tests on this part. If extracting the `exec.LookPath` part is not too difficult without overcomplicating it